docs/container-empty-disks.md

```diff
@@ -0,0 +1,83 @@
+# Provide temporary empty disks
+
+## Motivation
+
+KubeVirt in combination with use-cases like the cluster-autoscaler can be used
+to scale and provision vm pools. With the VirtualMachineReplicaSet and
+cloud-init stateless pools can be instantiated and provisioned.
+
+Many of the used cloud images for these use-cases are bootet with small disks
+with no additional free space, and/or run in read-only mode. A common pattern
+to store temporary data which should survive restarts or temporary data which
+is too big to fit in memory on temporary disks. The `emptyDir` in k8s provides
+a similar functionality.
+
+To provide a similar functionality for our `VirtualMachines` an `emptyDisk`
+volume type can be introduced.
+
+## Implementation
+
+A new volume type `emptyDisk` is introduced. The only parameter it supports is
+`capacity`. The `emptyDisk` will be placed by `virt-launcher` in an `emptyDir`,
+to make sure that the kubelet can take care of clean-up and that
+`ephemeral-store` resource definitions apply to the `emptyDisks` too.
+
+The definition look like this:
+
+
+```golang
+// EmptyDisk represents a temporary disk which shares the vms lifecycle
+type EmptyDiskSource struct {
+	// Capacity of the sparse disk.
+	Capacity resource.Quantity `json:"capacity"`
+}
+```
+
+A usage example:
+
+```yaml
+apiVersion: kubevirt.io/v1alpha1
+kind: VirtualMachine
+metadata:
+  name: testvm
+spec:
+  terminationGracePeriodSeconds: 0
+  domain:
+    resources:
+      requests:
+        memory: 512M
+        ephemeral-store: 5G
+      limits:
+        ephemeral-store: 7G
```

Comment:
can we not infer the disk store limits based on the capacity of the empty disk rather than exposing this to the user?

Just trying to understand the value this presents the user.

Reply:
I agree, will remove that.